utils/db.py

```diff
@@ -92,12 +93,14 @@ class Db(Mapping):
         tables, like the mapping interface or :py:meth:`values`.
 
     """
-    def __init__(self, hostname=None, credentials=None):
+    def __init__(self, hostname=None, credentials=None, db_port=None):
         self._table_cache = {}
-        if hostname is None:
+        if hostname is None and db_port is None:
```

Comment:
I'm looking at this and thinking that we should probably say:

If the hostname is ```None``` then the hostname and the port from current_appliance. I'm not sure we should allow the port to be set differently. A part of me wonders if we should just simply pass in the appliance object. I also think that perhaps we should amend the methods below to take an appliance object and not take a port/hostname combo. Thoughts?? @mfalesni ?